mantidimaging/core/filters/rotate_stack/rotate_stack.py

```diff
@@ -1,20 +1,84 @@
+from functools import partial
+
 import numpy as np
 
 from mantidimaging import helper as h
+from mantidimaging.core.filters.base_filter import BaseFilter
 from mantidimaging.core.parallel import shared_mem as psm
 from mantidimaging.core.parallel import utility as pu
 from mantidimaging.core.utility.progress_reporting import Progress
 
 
+class RotateFilter(BaseFilter):
+    filter_name = "Rotate Stack"
+
+    def _filter_func(self, data, rotation=None, flat=None, dark=None, cores=None, chunksize=None,
+                     progress=None):
+        """
+        Rotates a stack (sample, flat and dark images).
+
+        This function only works with square images.
+
+        If the picture is cropped first, the ROI coordinates have to be adjusted
+        separately to be pointing at the NON ROTATED image!
+
+        :param data: stack of sample images
+        :param rotation: The rotation to be performed
+        :param flat: flat images average
+        :param dark: dark images average
+        :param cores: cores for parallel execution
+        :param chunksize: chunk for each worker
+
+        :return: rotated images
+        """
+        h.check_data_stack(data)
+
+        if rotation:
+            # rot90 rotates counterclockwise; config.args.rotation rotates
+            # clockwise
+            clockwise_rotations = 4 - rotation
+
+            if pu.multiprocessing_available():
+                _execute_par(data, clockwise_rotations, cores, chunksize)
+            else:
+                _execute_seq(data, clockwise_rotations)
+
+            if flat is not None:
+                flat = _rotate_image(flat, clockwise_rotations)
+            if dark is not None:
+                dark = _rotate_image(dark, clockwise_rotations)
+
+        h.check_data_stack(data)
+
+        if flat is None and dark is None:
+            return data
+        else:
+            return data, flat, dark
+
+    def register_gui(self, form, on_change):
+        from mantidimaging.gui.utility import add_property_to_form
+
+        _, rotation_count = add_property_to_form(
+            'Number of rotations', 'int', 1, (0, 99),
+            form=form, on_change=on_change)
+
+        return {
+            "rotation_count": rotation_count
+        }
+
+    def execute_wrapper(self, rotation_count=None):
+        return partial(self._filter_func, rotation_count=rotation_count.value())
```

Comment:
I get an error here as I think `rotation_count` is trying to be passed into `rotation`, which then doesn't match the kwargs